### PR TITLE
Add support for running the profiler multiple times in a single request

### PR DESCRIPTION
**Pull request by mstyles**
The goal of this PR is to provide XHGui with the ability to arbitrarily start and stop the profiler at any point during run time. This is accomplished by refactoring the external/header.php file into a class, XHGui_Profiler, which provides the API for starting and stopping the configured profiler extension. The primary motivation for this enhancement comes from the desire to run multiple profiles in a single request (the application I'm working on, for example, is a long-running process that handles events over a socket connection).

I would really like to see this functionality added, even if the exact implementation ends up quite different from this. If you could please provide whatever questions or concerns you have, I will do whatever I can to make this possible. Thanks.

### CODE REVIEW

File: src/Xhgui/Db/Mapper.php [hunk 1]
```diff
@@ -57,10 +57,10 @@ protected function _conditions($search)
             $conditions['meta.simple_url'] = (string)$search['simple_url'];
         }
         if (!empty($search['request_start'])) {
-            $conditions['meta.SERVER.REQUEST_TIME']['$gte'] = $this->_convertDate($search['request_start']);
+            $conditions['meta.profile_ts']['$gte'] = $this->_convertDate($search['request_start']);
```
Comment (markstory) [inline src/Xhgui/Db/Mapper.php]: This will break existing installs as old data will not have this key. Can we add the new key in a more compatible way?

Comment (mstyles) [inline src/Xhgui/Db/Mapper.php]: Doh! Certainly, I will do that.

**Comment by lauripiisang** [comment]
Thank you @mstyles and sorry this has been on hold for so long ! 

My organization has similar functionality built on top of perftools/xhgui-collector and I hope we can get it published as OSS soon. I'll post back here as soon as that happens

Thanks again

**Comment by lauripiisang** [comment]
FWIW: https://github.com/perftools/xhgui-collector/pull/2 introduces a way to start/stop profiling sessions.

**Comment by Krinkle** [comment]
Given xhgui now uses xhgui-collector, does that mean this PR is now obsolete?

**Pull request closed without merging**